### PR TITLE
[16.0][ADD] sale_loyalty_initial_date_validity: New module

### PR DESCRIPTION
**Pull request by pilarvargas-tecnativa**
Pending from:

- [x] https://github.com/OCA/sale-promotion/pull/167

cc @Tecnativa TT45685

@chienandalu @ernesto-garcia-tecnativa please review

### CODE REVIEW

**Comment by ernesto-garcia-tecnativa** [comment]
@pilarvargas-tecnativa please apply rebase and remove the ```test-requirements.txt``` file.

File: sale_loyalty_initial_date_validity/tests/test_sale_loyalty_initial_date_validity.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+from odoo.exceptions import ValidationError
+from odoo.tests import Form, TransactionCase
+
+
+class TestSaleLoyaltyInitialDateValidity(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        loyalty_program = cls.env["loyalty.program"]
```
Comment (chienandalu) [inline sale_loyalty_initial_date_validity/tests/test_sale_loyalty_initial_date_validity.py:16]: If we don't check mail stuff let's disable it
```suggestion
        cls.env = cls.env(context=dict(cls.env.context, **DISABLED_MAIL_CONTEXT))
        loyalty_program = cls.env["loyalty.program"]
```

**Review by chienandalu — CHANGES_REQUESTED** [review]
Suggestion:

File: sale_loyalty_initial_date_validity/tests/test_sale_loyalty_initial_date_validity.py (hunk 1 above)
Comment (pilarvargas-tecnativa) [reply sale_loyalty_initial_date_validity/tests/test_sale_loyalty_initial_date_validity.py:16]: done :)

**Review by chienandalu — APPROVED** [review]
Thanks :+1:

**Comment by pilarvargas-tecnativa** [comment]
@CarlosRoca13 can you review please?

**Comment by chienandalu** [comment]
CIs are red

**Comment by chienandalu** [comment]
please @CarlosRoca13

**Review by CarlosRoca13 — APPROVED** [review]
Code review :+1:

**Comment by pilarvargas-tecnativa** [comment]
ping @pedrobaeza :)

**Pull request merged**